### PR TITLE
Optimise snitch GUI

### PR DESCRIPTION
**Pull request by okx-code**
- Only render the snitches for the current page
- Use components instead of adding text lore (this is about 90% faster than the previous method)
- Also fix the plugin name (previously it was jukealert-paper, now it correctly shows as JukeAlert)

### CODE REVIEW

**Comment by RedDevel2** [comment]
Now that I think about it. The name is probably why ItemExchange didn't recognize it. 💀

**Comment by okx-code** [comment]
Yep it broke for me as well

File: plugins/jukealert-paper/src/main/java/com/untamedears/jukealert/gui/SnitchOverviewGUI.java [hunk 1]
```diff
@@ -29,38 +38,39 @@ public SnitchOverviewGUI(Player player, List<Snitch> snitches, String title, boo
...
-				MetaUtils.addLore(meta, ChatColor.YELLOW + "Group: " + snitch.getGroup().getName());
+				List<Component> lore = new ArrayList<>();
+				lore.add(ChatUtils.newComponent("Location: ").color(NamedTextColor.AQUA)
+						.append(Component.text(location.getWorld().getName() + " ").color(NamedTextColor.WHITE))
+						.append(Component.text(location.getBlockX()).color(NamedTextColor.RED))
+						.append(Component.text(", ").color(NamedTextColor.WHITE))
+						.append(Component.text(location.getBlockY()).color(NamedTextColor.GREEN))
+						.append(Component.text(", ").color(NamedTextColor.WHITE))
+						.append(Component.text(location.getBlockZ()).color(NamedTextColor.BLUE)));
+				lore.add(ChatUtils.newComponent("Group: " + snitch.getGroup().getName()).color(NamedTextColor.YELLOW));
```
Comment (Protonull) [inline plugins/jukealert-paper/src/main/java/com/untamedears/jukealert/gui/SnitchOverviewGUI.java:58]: You should do the same with all of these components too :]

Comment (okx-code) [reply plugins/jukealert-paper/src/main/java/com/untamedears/jukealert/gui/SnitchOverviewGUI.java:58]: Nah

**Comment by Protonull** [comment]
> Now that I think about it. The name is probably why ItemExchange didn't recognize it. 💀

That could be the cause of some unintended plugin-load ordering, but it should NOT cause the error. JukeAlert is supposed to be a soft-dependency; ItemExchange is suppose to work even if JukeAlert isn't there.

The error is also completely baffling, it's a `NoClassDefFoundError`, which means a class is being used in a non-reflective manner (eg: an instance is being constructed with `new`), but the class isn't present.

The missing class is `LoggableAction`, right?. That by itself is bizarre because nothing in ItemExchange references `LoggableAction`: the class `ShopPurchaseAction` extends `LoggablePlayerAction` and *that* implements `LoggableAction`, so... why is the missing class `LoggableAction` and not `LoggablePlayerAction`? Those are both JukeAlert classes.

Also, `ShopPurchaseAction` is only interacted with in two places:

1. Inside a lambda that's registered as an action-factory to JukeAlert, something that only happens if JukeAlert is enabled.

2. Inside an event listener that's only registered if JukeAlert is enabled.

So how in the hell is this class being used?!

**Comment by Protonull** [comment]
Ohhhh, okay, just figured out what's happening: the action-factory lambda gets hoisted by javac into a sibling method:
image

So when the class is evaluated, it finds the method and tries to evaluate its return type, LoggableAction, which isn't loaded. The only fix to this is to deliberately avoid the lambda-isation and turn it into a sibling class.

**Pull request merged**